apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common/BufferOptions.cs

```diff
@@ -0,0 +1,93 @@
+﻿// Copyright 2016 Google Inc. All Rights Reserved.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+
+namespace Google.Cloud.Diagnostics.Common
+{
+    /// <summary>
+    /// Types of buffers.
+    /// </summary>
+    public enum BufferType
+    {
+        /// <summary>No buffer will be use and the base consumer will be returned.</summary>
+        None,
+
+        /// <summary>A sized buffer will be used to wrap the base consumer.</summary>
+        Sized,
+
+        /// <summary>A timed buffer will be used to wrap the base consumer.</summary>
+        Timed
+    }
+
+    /// <summary>
+    /// Options that define a buffer.
+    /// </summary>
+    public class BufferOptions
+    {
+        /// <summary>The default buffer size in bytes. 2^16 = 65536.</summary>
+        private const int DefaultBufferSize = 65536;
+
+        /// <summary>The default amount of time to wait before a flush is automatically made.</summary>
+        private static readonly TimeSpan DefaultWaitTime = TimeSpan.FromSeconds(5);
+
+        /// <summary>The type of buffer to be used.</summary>
+        public BufferType BufferType { get; private set; }
+
+        /// <summary>The size of the buffer in bytes, only used for <see cref="BufferType.Sized"/></summary>
+        public int BufferSizeBytes { get; private set; }
+
+        /// <summary>The time to wait before the buffer is flushed, only used for <see cref="BufferType.Timed"/></summary>
+        public TimeSpan BufferWaitTime { get; private set; }
+
+        private BufferOptions() { }
+
+        /// <summary>
+        /// Create <see cref="BufferOptions"/> for <see cref="BufferType.None"/>
```

Comment:
When using `<see .../>` references like this, the API docs will have a link to just the name of the enum member, so this (and similar summaries) will say "Create BufferOptions for None". I think something like this would be clearer:

```csharp
/// Gets a <see cref="BufferOptions"/> instance to describe a <see cref="BufferType.None"/> type of buffer.
```

which is admittedly a little awkward for the None value, but works better for other values.